source/extensions/filters/http/common/aws/utility.cc

```diff
@@ -0,0 +1,54 @@
+#include "extensions/filters/http/common/aws/utility.h"
+
+#include "common/common/fmt.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+std::map<std::string, std::string> Utility::canonicalizeHeaders(const Http::HeaderMap& headers) {
+  std::map<std::string, std::string> out;
+  headers.iterate(
+      [](const Http::HeaderEntry& entry, void* context) -> Http::HeaderMap::Iterate {
+        auto* map = static_cast<std::map<std::string, std::string>*>(context);
+        // Pseudo-headers should not be canonicalized
+        if (entry.key().c_str()[0] == ':') {
```

Comment:
Maybe an empty header in there somehow as well?

Reply:
Sure